packages/flutter/lib/src/material/page.dart

```diff
@@ -88,12 +89,25 @@ class AnimationMean extends CompoundAnimation<double> {
 // is fully on-screen. 0.0 is fully off-screen to the right. 1.0 is off-screen
 // to the left.
 class _CupertinoTransitionCurve extends Curve {
-  _CupertinoTransitionCurve();
+  _CupertinoTransitionCurve(this.curve);
+
+  Curve curve;
 
   @override
   double transform(double t) {
-    if (t > 0.5)
-      return (t - 0.5) / 3.0 + 0.5;
+    // The range is divided in 2 halves: [0.0, 0.5] applies to the top-most
+    // route, while [0.5, 1.0] applies to the one underneath it. We scale both
+    // ranges to [0.0, 1.0] when applying transformations.
```

Comment:
I don't understand why this is true. Suppose you popped a route, then 20ms later popped another. Both routes would be moving at the same time. There's no guarantee that only one of the animations will be changing at any particular time.